metadata/src/main/java/org/apache/kafka/controller/BrokerHeartbeatManager.java

```diff
@@ -276,7 +277,7 @@ Collection<BrokerHeartbeatState> brokers() {
     }
 
     // VisibleForTesting
-    long getControlledShutDownOffset(int brokerId) {
+    long controlledShutDownOffset(int brokerId) {
```

Comment:
nit: could we call this `controlledShutdownOffset`. I think it is a bit more conventional to see "shutdown" as one word, as in `updateControlledShutdownOffset` below. (Similarly, perhaps we can rename the field.)

Also, how about returning `OptionalLong` from this method to make it a bit safer? We can return `OptionalLong.empty()` if either the brokerId is not present in `brokers` or if `controlledShutdownOffset` is -1.

Reply:
hm, I think changing `controlledShutDownOffset()` to return `OptionalLong.empty()` instead of `-1` could be slightly unintuitive as `-1` was specifically allocated to mean the broker isn't currently in controlled shutdown. I think it would make more sense if the underlying field was also type `OptionalLong` and used `OptionalLong.empty()` instead of `-1`

Reply:
Is your concern that `Optional.empty` would be used to handle two situations? Perhaps we could raise an exception if the `brokerId` is not present in `brokers`? Then `Optional.empty` would be reserved only for the case when the broker is not in controlled shutdown.